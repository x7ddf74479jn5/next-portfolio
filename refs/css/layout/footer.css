.footer {
  padding-bottom: 0;
  padding-top: 0;
}
.footer a:hover, .footer .footer a:hover * {
  color: #ffb549;
}
.footer ul {
  list-style: none;
}
.footer .main {
  color: #4c4c4c;
  text-align: left;
}
.footer .main [dir=rtl] {
  text-align: right;
}
.footer .main .main-inner {
  margin: 0 auto;
  max-width: 1440px;
  padding: 0;
}
@media screen and (min-width: 576px) {
  .footer .main .main-inner {
    padding: 0 32px;
  }
}
.footer .main .logo-area {
  padding: 2rem 0;
}
.footer .main .logo-area .product-logo {
  line-height: 1.875rem;
  margin: 0 auto 1rem auto;
  text-align: center;
}
.footer .main .logo-area .product-logo .logo-img {
  height: 3rem;
  width: 6rem;
}
@media screen and (min-width: 1024px) {
  .footer .main .logo-area {
    float: left;
    padding: 0;
  }
  .footer .main .logo-area .product-logo {
    text-align: left;
  }
  .footer .main .logo-area .product-logo .logo-img {
    height: 4rem;
    width: 8rem;
  }
}
.footer .main .sitemap {
  padding: 0;
}
.footer .main .sitemap .footer-links {
  margin: 0;
}
.footer .main .sitemap .footer-links ul {
  list-style: none;
}
@media screen and (min-width: 1024px) {
  .footer .main .sitemap .footer-links {
    display: flex;
    border-top: none;
    justify-content: flex-end;
  }
}
.footer .main .foot {
  padding-top: 1rem;
}
.footer .main .foot ul {
  margin: 0.5rem 1rem;
}
.footer .main .foot ul a {
  color: #777;
  text-decoration: none;
}
.footer .main .foot ul a:visited {
  color: #777;
}
.footer .main .foot ul a li {
  display: inline-block;
  font-size: 0.9rem;
  font-weight: 600;
  margin-right: 1rem;
}
@media screen and (min-width: 1024px) {
  .footer .main .foot {
    -ms-flex-align: center;
    align-items: center;
    border-top: 2px solid #4c4c4c;
    display: -ms-flexbox;
    display: flex;
    padding-top: 2rem;
  }
  .footer .main .foot li {
    margin-top: 0.125rem;
  }
}
