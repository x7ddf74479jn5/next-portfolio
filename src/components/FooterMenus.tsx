import Image from "next/image";
import Link from "next/link";
import { links as l } from "src/static/links";

const links = {
  about: [
    { label: "Vision", href: "/about/#vision" },
    { label: "Profile", href: "/about/#profile" },
    { label: "Career", href: "/about/#career" },
    { label: "Skills", href: "/about/#skills" },
  ],
  samples: [
    { label: "ポートフォリオサイト", href: "/samples/portfolio/" },
    { label: "ECサイト", href: "/samples/ec/" },
    { label: "Chat App", href: "/samples/chat/" },
  ],
  links: [
    { label: "Blog", href: "/" },
    { label: "Twitter", href: l.twitter },
    // {label: "",href:""},
    // {label: "",href:""},
  ],
};
const accordions = [
  { label: "自己紹介", id: "footer-sitemap-about", inputId: "accordion-about", links: links.about },
  { label: "制作事例", id: "footer-sitemap-works", inputId: "accordion-works", links: links.samples },
  { label: "リンク集", id: "footer-sitemap-links", inputId: "accordion-links", links: links.links },
];

export const FooterMenus = () => {
  const handle = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
    console.log(accordionModule.toggleMenu("accordion-about"));
    console.log(accordionModule.toggleMenu("accordion-works"));
  };

  const accordionItems = accordions.map((accordion) => (
    <li className="p-accordion" id={accordion.id} key={accordion.id}>
      {/* <button onClick={handle} onKeyDown={(e) => handle}> */}
      <h4 role="button" onClick={handle} onKeyDown={(e) => handle}>
        {accordion.label}
      </h4>
      {/* </button> */}
      <input type="checkbox" id={accordion.inputId} />
      <label htmlFor={accordion.inputId}></label>
      <ul className="p-accordion__items">
        {accordion.links.map((link, index) => (
          <Link href={link.href} key={index}>
            <a target="_self">
              <li>{link.label}</li>
            </a>
          </Link>
        ))}
        {/* {accordion.links.map((link, index) => {
          <li key={index}>
            <Link href={link.href}>
              <a target="_self">{link.label}</a>
            </Link>
          </li>
        })} */}
      </ul>
    </li>
  ));

  return (
    <section className="sitemap">
      <ul className="footer-links">
        {accordionItems}
        {/* <li className="p-accordion" id="footer-sitemap-about">
        <h4 role="button" onClick={handle}>
          自己紹介
        </h4>
        <input type="checkbox" id="accordion-about" />
        <label htmlFor="accordion-about"></label>
        <ul className="p-accordion__items">
          <a href="/about/#vision" target="_self">
            <li>Vision</li>
          </a>
          <a href="/about/#profile" target="_self">
            <li>Profile</li>
          </a>
          <a href="/about/#career" target="_self">
            <li>Career</li>
          </a>
          <a href="/about/#skills" target="_self">
            <li>Skills</li>
          </a>
        </ul>
      </li>
      <li className="p-accordion" id="footer-sitemap-works">
        <h4 role="button" onClick={handle}>
          制作事例
        </h4>
        <input type="checkbox" id="accordion-works" />
        <label htmlFor="accordion-works"></label>
        <ul className="p-accordion__items">
          <a href="/sample/portfolio/" target="_self">
            <li>ポートフォリオサイト</li>
          </a>
          <a href="/sample/ec/" target="_self">
            <li>ECサイト</li>
          </a>
          <a href="/sample/chat/" target="_self">
            <li>Chat App</li>
          </a>
        </ul>
      </li>
      <li className="p-accordion" id="footer-sitemap-links">
        <h4 role="button" onClick={handle}>
          リンク集
        </h4>
        <input type="checkbox" id="accordion-links" />
        <label htmlFor="accordion-links"></label>
        <ul className="p-accordion__items">
          <a href="#" target="_self">
            <li>Blog</li>
          </a>
          <a href="#" target="_self">
            <li>Twitter</li>
          </a>
          <a href="#" target="_blank">
            <li>YouTube</li>
          </a>
          <a href="#" target="_blank">
            <li>Qiita</li>
          </a>
          <a href="#" target="_blank">
            <li>GitHub</li>
          </a>
        </ul>
      </li> */}
      </ul>
    </section>
  );
};
