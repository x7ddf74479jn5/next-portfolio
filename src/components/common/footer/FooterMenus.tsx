import Link from "next/link";
import { externalLinks, links } from "src/utils/paths";

const accordionItems = {
  about: [
    { label: "Vision", href: links.about + "#vision" },
    { label: "Profile", href: links.about + "#profile" },
    { label: "Career", href: links.about + "#career" },
    { label: "Skills", href: links.about + "#skills" },
  ],
  samples: [
    { label: "ポートフォリオサイト", href: links.portfolio },
    { label: "ECサイト", href: links.ec },
    { label: "Chat App", href: links.chat },
  ],
  links: [
    { label: "Twitter", href: externalLinks.twitter },
    { label: "Github", href: externalLinks.github },
  ],
};
const accordions = [
  { label: "自己紹介", id: "footer-sitemap-about", inputId: "accordion-about", links: accordionItems.about },
  { label: "制作事例", id: "footer-sitemap-works", inputId: "accordion-works", links: accordionItems.samples },
  { label: "リンク集", id: "footer-sitemap-links", inputId: "accordion-links", links: accordionItems.links },
];

export const FooterMenus: React.FC = () => {
  const handle = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {};

  const accordionListItems = accordions.map((accordion) => {
    return (
      <li className="p-accordion" id={accordion.id} key={accordion.id}>
        {/* <button onClick={handle} onKeyDown={(e) => handle}> */}
        <h4
          role="button"
          onClick={handle}
          onKeyDown={(e) => {
            return handle;
          }}
        >
          {accordion.label}
        </h4>
        {/* </button> */}
        <input type="checkbox" id={accordion.inputId} />
        <label htmlFor={accordion.inputId}></label>
        <ul className="p-accordion__items">
          {accordion.links.map((link, index) => {
            return (
              <Link href={link.href} key={index}>
                <a target="_self">
                  <li>{link.label}</li>
                </a>
              </Link>
            );
          })}
          {/* {accordion.links.map((link, index) => {
          <li key={index}>
            <Link href={link.href}>
              <a target="_self">{link.label}</a>
            </Link>
          </li>
        })} */}
        </ul>
      </li>
    );
  });

  return (
    <section className="sitemap">
      <ul className="footer-links">
        {accordionListItems}
        {/* <li className="p-accordion" id="footer-sitemap-about">
        <h4 role="button" onClick={handle}>
          自己紹介
        </h4>
        <input type="checkbox" id="accordion-about" />
        <label htmlFor="accordion-about"></label>
        <ul className="p-accordion__items">
          <a href="/about/#vision" target="_self">
            <li>Vision</li>
          </a>
          <a href="/about/#profile" target="_self">
            <li>Profile</li>
          </a>
          <a href="/about/#career" target="_self">
            <li>Career</li>
          </a>
          <a href="/about/#skills" target="_self">
            <li>Skills</li>
          </a>
        </ul>
      </li>
      <li className="p-accordion" id="footer-sitemap-works">
        <h4 role="button" onClick={handle}>
          制作事例
        </h4>
        <input type="checkbox" id="accordion-works" />
        <label htmlFor="accordion-works"></label>
        <ul className="p-accordion__items">
          <a href="/sample/portfolio/" target="_self">
            <li>ポートフォリオサイト</li>
          </a>
          <a href="/sample/ec/" target="_self">
            <li>ECサイト</li>
          </a>
          <a href="/sample/chat/" target="_self">
            <li>Chat App</li>
          </a>
        </ul>
      </li>
      <li className="p-accordion" id="footer-sitemap-links">
        <h4 role="button" onClick={handle}>
          リンク集
        </h4>
        <input type="checkbox" id="accordion-links" />
        <label htmlFor="accordion-links"></label>
        <ul className="p-accordion__items">
          <a href="#" target="_self">
            <li>Blog</li>
          </a>
          <a href="#" target="_self">
            <li>Twitter</li>
          </a>
          <a href="#" target="_blank">
            <li>YouTube</li>
          </a>
          <a href="#" target="_blank">
            <li>Qiita</li>
          </a>
          <a href="#" target="_blank">
            <li>GitHub</li>
          </a>
        </ul>
      </li> */}
      </ul>
    </section>
  );
};
