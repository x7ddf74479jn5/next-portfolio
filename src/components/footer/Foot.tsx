import Link from "next/link";

type Props = {
  href: string;
  label: string;
};

const Foot: React.VFC<Props> = ({ href, label }) => {
  return (
    <Link href={href}>
      <a>{label}</a>
    </Link>
  );
};

export default Foot;
