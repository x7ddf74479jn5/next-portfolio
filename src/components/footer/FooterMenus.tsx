import Link from "next/link";
import React from "react";
import styles from "src/styles/components/footer/Footer.module.scss";
import { externalLinks, links } from "src/utils/paths";

import useMedia from "../../hooks/useMedia";
import { identity } from "../../utils/helper";

const accordions = identity({
  about: {
    label: "自己紹介",
    items: [
      { label: "Vision", href: links.about + "#vision" },
      { label: "Profile", href: links.about + "#profile" },
      { label: "Career", href: links.about + "#career" },
      { label: "Skills", href: links.about + "#skills" },
    ],
  },
  samples: {
    label: "制作事例",
    items: [
      { label: "ポートフォリオサイト", href: links.portfolio },
      { label: "ECサイト", href: links.ec },
      { label: "Chat App", href: links.chat },
    ],
  },
  links: {
    label: "リンク集",
    items: [
      { label: "Twitter", href: externalLinks.twitter },
      { label: "Github", href: externalLinks.github },
    ],
  },
});

type Accordion = {
  label: string;
  items: Array<{ label: string; href: string }>;
};

const Accordion: React.VFC<Accordion> = (accordion) => {
  const { isPC } = useMedia();

  return (
    <details open={isPC}>
      <summary>{accordion.label}</summary>
      <ul className={styles.accordionItems}>
        {accordion.items.map((item, index) => {
          return (
            <Link href={item.href} key={index}>
              <li>
                <a target="_self">{item.label}</a>
              </li>
            </Link>
          );
        })}
      </ul>
    </details>
  );
};

export const FooterMenus: React.VFC = () => {
  return (
    <section className={styles.sitemap}>
      <ul className={styles.footerLinks}>
        <li className={styles.accordion}>
          {<Accordion label={accordions.about.label} items={accordions.about.items} />}
        </li>
        <li className={styles.accordion}>
          {<Accordion label={accordions.samples.label} items={accordions.samples.items} />}
        </li>
        <li className={styles.accordion}>
          {<Accordion label={accordions.links.label} items={accordions.links.items} />}
        </li>
      </ul>
    </section>
  );
};
