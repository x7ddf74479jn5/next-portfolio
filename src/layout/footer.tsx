import Image from "next/image";
import Link from "next/link";
import { FooterMenus } from "src/components/FooterMenus";

export const Footer: React.FC = () => {
  // <footer>
  //   <small lang="en">@ 2020 example.com</small>
  // </footer>

  const handle = (event: React.MouseEvent<HTMLHeadingElement, MouseEvent>) => {
    console.log(accordionModule.toggleMenu("accordion-about"));
    console.log(accordionModule.toggleMenu("accordion-works"));
  };

  return (
    <footer className="footer">
      <div className="main c-section">
        <div className="main-inner">
          <section className="logo-area">
            <h1 className="product-logo">
              <Link href="/">
                <a target="_self">
                  {/* <Image height={} width={} alt="パンダシャーク" className="logo-img" src="/img/icons/pandashark_logo.png" /> */}
                  <img alt="パンダシャーク" className="logo-img" src="/img/icons/pandashark_logo.png" />
                </a>
              </Link>
              {/* <a href="/" target="_self">
                <img alt="パンダシャーク" className="logo-img" src="/img/icons/pandashark_logo.png" />
              </a> */}
            </h1>
            {/* <a className="p-btn-round p-icon__arrow-next" href="/community/" target="_self" role="button">
              Webを学びたい？
            </a> */}
          </section>
          <FooterMenus />

          {/* <section className="sitemap">
            <ul className="footer-links">
              <li className="p-accordion" id="footer-sitemap-about">
                <h4 role="button" onClick={handle}>
                  自己紹介
                </h4>
                <input type="checkbox" id="accordion-about" />
                <label htmlFor="accordion-about"></label>
                <ul className="p-accordion__items">
                  <a href="/about/#vision" target="_self">
                    <li>Vision</li>
                  </a>
                  <a href="/about/#profile" target="_self">
                    <li>Profile</li>
                  </a>
                  <a href="/about/#career" target="_self">
                    <li>Career</li>
                  </a>
                  <a href="/about/#skills" target="_self">
                    <li>Skills</li>
                  </a>
                </ul>
              </li>
              <li className="p-accordion" id="footer-sitemap-works">
                <h4 role="button" onClick={handle}>
                  制作事例
                </h4>
                <input type="checkbox" id="accordion-works" />
                <label htmlFor="accordion-works"></label>
                <ul className="p-accordion__items">
                  <a href="/sample/portfolio/" target="_self">
                    <li>ポートフォリオサイト</li>
                  </a>
                  <a href="/sample/ec/" target="_self">
                    <li>ECサイト</li>
                  </a>
                  <a href="/sample/chat/" target="_self">
                    <li>Chat App</li>
                  </a>
                </ul>
              </li>
              <li className="p-accordion" id="footer-sitemap-links">
                <h4 role="button" onClick={handle}>
                  リンク集
                </h4>
                <input type="checkbox" id="accordion-links" />
                <label htmlFor="accordion-links"></label>
                <ul className="p-accordion__items">
                  <a href="#" target="_self">
                    <li>Blog</li>
                  </a>
                  <a href="#" target="_self">
                    <li>Twitter</li>
                  </a>
                  <a href="#" target="_blank">
                    <li>YouTube</li>
                  </a>
                  <a href="#" target="_blank">
                    <li>Qiita</li>
                  </a>
                  <a href="#" target="_blank">
                    <li>GitHub</li>
                  </a>
                </ul>
              </li>
            </ul>
          </section> */}

          <section className="foot">
            <ul>
              <Link href="/terms/">
                <a>
                  <li>利用規約</li>
                </a>
              </Link>
              <Link href="/privacy/">
                <a>
                  <li>プライバシーポリシー</li>
                </a>
              </Link>

              {/* <a href="/terms/"><li>利用規約</li></a>
            <a href="/privacy/"><li>プライバシーポリシー</li></a> */}
            </ul>
          </section>
        </div>
      </div>
    </footer>
  );
};
