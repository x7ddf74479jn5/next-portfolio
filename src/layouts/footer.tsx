// import Image from "next/image";
import Link from "next/link";
import { FooterMenus } from "src/components/common/footer/FooterMenus";
import PandasharkLogo from "src/components/common/PandasharkLogo";
import styles from "src/styles/components/common/PandasharkLogo.module.scss";

export const Footer: React.FC = () => {
  const handle = (event: React.MouseEvent<HTMLHeadingElement, MouseEvent>) => {};

  return (
    <footer className="footer">
      <div className="main c-section">
        <div className="main-inner">
          <section className={styles.logoArea}>
            <PandasharkLogo width={128} height={64} />
            {/* <h1 className={styles.productlogo}>
              <Link href="/">
                <a target="_self">
                  <Image
                    width={128}
                    height={64}
                    alt="パンダシャーク"
                    className="logo-img"
                    src="/img/icons/pandashark_logo_rectangle.png"
                  />
                </a>
              </Link>
            </h1> */}

            {/* <a href="/" target="_self">
                <img alt="パンダシャーク" className="logo-img" src="/img/icons/pandashark_logo.png" />
              </a> */}
            {/* <a className="p-btn-round p-icon__arrow-next" href="/community/" target="_self" role="button">
              Webを学びたい？
            </a> */}
          </section>
          <FooterMenus />

          <section className="foot">
            <ul>
              <Link href="/terms/">
                <a>
                  <li>利用規約</li>
                </a>
              </Link>
              <Link href="/privacy/">
                <a>
                  <li>プライバシーポリシー</li>
                </a>
              </Link>

              {/* <a href="/terms/"><li>利用規約</li></a>
            <a href="/privacy/"><li>プライバシーポリシー</li></a> */}
            </ul>
          </section>
        </div>
      </div>
    </footer>
  );
};
