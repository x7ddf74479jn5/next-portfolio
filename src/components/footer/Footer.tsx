import PandasharkLogo from "src/components/common/PandasharkLogo";
import Foot from "src/components/footer/Foot";
import { FooterMenus } from "src/components/footer/FooterMenus";
import styles from "src/styles/components/footer/Footer.module.scss";

import { internalLinks } from "../../utils/paths";

export type FootItem = {
  id: string;
  href: string;
  label: string;
};

const footItems: FootItem[] = [
  { id: "terms", href: internalLinks.terms, label: "利用規約" },
  { id: "privacy", href: internalLinks.privacy, label: "プライバシーポリシー" },
];

export const Footer: React.VFC = () => {
  return (
    <footer className={styles.footer}>
      <div className={styles.main}>
        <div className={styles.mainInner}>
          <div className={styles.wrapper}>
            <section className={styles.logoArea}>
              <div className={styles.productLogo}>
                <PandasharkLogo width={128} height={64} />
              </div>
            </section>
            <FooterMenus />
          </div>
          <Foot footItems={footItems} />
        </div>
      </div>
    </footer>
  );
};
