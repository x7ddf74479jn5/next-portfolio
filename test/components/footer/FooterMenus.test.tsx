import userEvent from "@testing-library/user-event";
import { Accordion, FooterMenus } from "src/components/footer/FooterMenus";
import type { RenderResult } from "test/test-utils";
import { render, setupWindow } from "test/test-utils";

describe("FooterMenus", () => {
  let renderResult: RenderResult;
  setupWindow();

  beforeEach(() => {
    renderResult = render(<FooterMenus />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    const detailsElement = renderResult.getAllByRole("group");
    expect(detailsElement).toHaveLength(3);
    const summaryElement = renderResult.container.getElementsByTagName("summary");
    expect(summaryElement).toHaveLength(3);
    const listElement = renderResult.getAllByRole("list")[0];
    expect(listElement.children).toHaveLength(3);
  });
});

describe("Accordion", () => {
  let renderResult: RenderResult;
  setupWindow();
  const accordionItem = {
    label: "summary",
    items: [
      {
        label: "label 1",
        href: "href 1",
      },
      {
        label: "label 2",
        href: "href 2",
      },
      {
        label: "label 3",
        href: "href 3",
      },
    ],
  };

  beforeEach(() => {
    renderResult = render(<Accordion accordion={accordionItem} />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    const summaryElement = renderResult.getByText("summary");
    expect(summaryElement).toBeInTheDocument();

    const listElement = renderResult.getByRole("list");
    expect(listElement.children).toHaveLength(3);

    const listItemElement = renderResult.getAllByRole("listitem");
    expect(listItemElement[0]).toHaveTextContent("label");

    const anchorElement = renderResult.getAllByRole("link");
    expect(anchorElement).toHaveLength(3);
    expect(anchorElement[0]).toHaveAttribute("href", expect.stringMatching("href"));
  });

  describe("user interaction", () => {
    let detailsElement: HTMLDetailsElement;
    let summaryElement: HTMLElement;

    it("opens when clicked", () => {
      detailsElement = renderResult.getByRole("group") as HTMLDetailsElement;
      summaryElement = renderResult.getByText("summary");

      expect(detailsElement.open).not.toBe(true);
      userEvent.click(summaryElement);
      expect(detailsElement.open).toBe(true);
    });

    it("nextly closes when clicked", () => {
      expect(detailsElement.open).toBe(true);
      userEvent.click(summaryElement);
      expect(detailsElement.open).not.toBe(true);
    });
  });
});
