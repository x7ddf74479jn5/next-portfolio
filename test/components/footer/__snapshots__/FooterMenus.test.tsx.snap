// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion matches snapshot 1`] = `
<DocumentFragment>
  <details
    class="accordion"
  >
    <summary>
      summary
    </summary>
    <ul
      class="accordionItems"
    >
      <li>
        <a
          href="/href%201"
        >
          label 1
        </a>
      </li>
      <li>
        <a
          href="/href%202"
        >
          label 2
        </a>
      </li>
      <li>
        <a
          href="/href%203"
        >
          label 3
        </a>
      </li>
    </ul>
  </details>
</DocumentFragment>
`;

exports[`FooterMenus matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="sitemap"
  >
    <ul
      class="footerLinks"
    >
      <li>
        <details
          class="accordion"
        >
          <summary>
            自己紹介
          </summary>
          <ul
            class="accordionItems"
          >
            <li>
              <a
                href="/about#vision"
              >
                Vision
              </a>
            </li>
            <li>
              <a
                href="/about#profile"
              >
                Profile
              </a>
            </li>
            <li>
              <a
                href="/about#career"
              >
                Career
              </a>
            </li>
            <li>
              <a
                href="/about#skills"
              >
                Skills
              </a>
            </li>
          </ul>
        </details>
      </li>
      <li>
        <details
          class="accordion"
        >
          <summary>
            制作事例
          </summary>
          <ul
            class="accordionItems"
          >
            <li>
              <a
                href="/samples/portfolio"
              >
                ポートフォリオサイト
              </a>
            </li>
            <li>
              <a
                href="/samples/my-favorite-gear"
              >
                My Favorite Gear
              </a>
            </li>
            <li>
              <a
                href="/samples/food-blog"
              >
                Food Blog
              </a>
            </li>
            <li>
              <a
                href="/samples/ec"
              >
                ECサイト
              </a>
            </li>
            <li>
              <a
                href="/samples/chat"
              >
                Chat App
              </a>
            </li>
          </ul>
        </details>
      </li>
      <li>
        <details
          class="accordion"
        >
          <summary>
            リンク集
          </summary>
          <ul
            class="accordionItems"
          >
            <li>
              <a
                href="https://twitter.com/pandashark6"
              >
                Twitter
              </a>
            </li>
            <li>
              <a
                href="https://github.com/x7ddf74479jn5"
              >
                Github
              </a>
            </li>
          </ul>
        </details>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
